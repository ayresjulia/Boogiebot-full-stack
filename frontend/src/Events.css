.Events-body {
	margin-top: 0px;
	margin-left: auto;
	margin-right: auto;
	display: grid;
	grid-template-columns: 30% 70%;
	text-align: center;
	justify-content: space-evenly;
	width: 100%;
	border-left: 1px solid rgb(238, 238, 238);
	height: auto;
}

.Events-create {
	font-size: 1.5rem;
	display: inline-block;
	text-align: center;
	margin-top: 40%;
}

.Events-new {
	margin: 20px;
	transition: all .2s ease-in-out;
	text-decoration: none;
	color: rgb(53, 52, 52);
	background-color: rgba(37, 36, 216, 0.5);
}

.big-bold {
	font-size: 80px;
}

.Events-current {
	display: grid;
	grid-template-columns: auto auto;
	margin-left: auto;
	margin-right: auto;
	text-align: center;
	justify-content: space-evenly;
	width: auto;
	min-height: 600px;
}

.evt {
	margin: 10px;
	transition: all .2s ease-in-out;
	color: rgb(53, 52, 52);
	background-color: rgb(246, 249, 252) !important;
	width: 250px;
}

.evt-img {
	height: 200px;
}

.Events-new:hover,
.evt:hover {
	transform: scale(1.1);
	-webkit-box-shadow: 0 20px 20px rgb(153, 154, 158);
	-moz-box-shadow: 0 20px 20px rgb(153, 154, 158);
	box-shadow: 0 20px 20px rgb(153, 154, 158);
	color: rgb(53, 52, 52);
}

.link {
	text-decoration: none;
}

.Events-rmrk {
	margin-top: 20px;
	text-align: center;
}
