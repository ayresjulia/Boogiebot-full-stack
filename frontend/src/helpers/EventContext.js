import React from "react";

/** Context: provides events object and setter for it throughout app. */

const EventContext = React.createContext();

export default EventContext;
