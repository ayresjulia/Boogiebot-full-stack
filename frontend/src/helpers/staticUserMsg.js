const staticMsg = {
	FORM_TIP: "tip.",
	EDIT_PROFILE_FORM_TIP_DESC:
		"while updating your info, keep in mind, that picking a great playlist is key for any party!",
	EDIT_PROFILE_FORM_CONFIRM_PWD: "Confirm password to make changes",
	FORM_USERNAME: "Username",
	FORM_PWD: "Password",
	FORM_FIRST_NAME: "First Name",
	FORM_LAST_NAME: "Last Name",
	FORM_EMAIL: "Email",
	FORM_PROFILE_URL: "Profile URL",
	EVENT_FORM_TIP_DESC:
		"scoop your ice-cream into a cupcake pan and pop in the freezer for a hassle-free	serving at your party.",
	LOGO_NAME: "BoogieBot",
	NAV_MY_EVENTS: "My Events",
	NAV_INSPIRATION: "Get Inspired",
	NAV_CATERING: "Catering",
	NAV_PROFILE: "Profile",
	NAV_LOG_OUT: "Log Out",
	NAV_LOG_IN: "Log In",
	NAV_SIGN_UP: "Sign Up",
	LOADING: "Loading...",
	CATERING_LOOKUP: "lookup",
	CATERING_FOOTNOTE:
		"check the event you want to save the restaurant to, it will appear in your event details",
	CATERING_INFO: "restaurants nearby with your zip code for catering options.",
	EVENTS_CREATE: "create",
	EVENTS_NEW: "new event.",
	HOME_WELCOME_USER: "Welcome, ",
	NON_USER_MSG: "Please, Log in or Sign up to begin.",
	MY_EVENT_DESC: "Description:",
	MY_EVENT_WHEN: "When:",
	MY_EVENT_TIME: "Time:",
	MY_EVENT_LOCATION: "Location:",
	DEFAULT_EVENT_IMG:
		"https://images.unsplash.com/photo-1532117182044-031e7cd916ee?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80",
	INSPIRATION_FOOTNOTE:
		"check the event you want to save the picture to, it will appear in your event details",
	INSPIRATION_ADD: "add",
	INSPIRATION_INFO: "pictures to your event moodboard for inspiration.",
	HOME_IMG1:
		"https://images.unsplash.com/photo-1620324446278-be6c7bb3ece5?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80",
	HOME_IMG2:
		"https://images.unsplash.com/photo-1527275393322-8ddae8bd5de9?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80",
	HOME_IMG3:
		"https://images.unsplash.com/photo-1525268771113-32d9e9021a97?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80",
	HOME_IMG4:
		"https://images.unsplash.com/photo-1534119238569-839b7389d552?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=800&q=80",
	CONSOLE_EVENTS_ERROR: "ERROR!!! Events could not be loaded",
	CONSOLE_USER_ERROR: "ERROR!!! currentUser.username could not be loaded",
	MY_EVENT_CAT: "catering",
	CATERING_DEFAULT:
		"https://images.unsplash.com/photo-1557962011-4b36a6f5d921?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1350&q=80",
	CATERING_WEB: "website",
	USER_DEFAULT_URL: "https://supersimple.com/wp-content/uploads/peek-a-boo-800-800-200x200.jpg",
	API_ERROR: "There is an error with getting data from API. Please check connection",
	EDIT_EVENT_FORM_TIP_DESC:
		"don't forget about younger audience, set up a cool kids table at your party",
	EDIT_EVENT_FORM_TITLE: "Title",
	EDIT_EVENT_FORM_DESC: "Description",
	EDIT_EVENT_FORM_EVENT_DATE: "Event Date",
	EDIT_EVENT_FORM_EVENT_TIME: "Event Time",
	EDIT_EVENT_FORM_CITY: "City",
	MY_EVENT_RMRK1: "browse pictures in Get Inspired tab to save them here",
	MY_EVENT_RMRK2: "browse restaurants in Catering tab to save them here",
	EVENTS_RMRK:
		"your events will be shown here. click on event to get details or create a new one",
	TARGET_ERR: "targetEvent could not be loaded",
	EVENT_ALERT: "Please choose an event first."
};

export default staticMsg;
