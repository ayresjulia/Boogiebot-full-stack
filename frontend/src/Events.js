import React, { useContext, useEffect } from "react";
import { Card, CardImg, CardBody, CardTitle, Button } from "reactstrap";
import { Link } from "react-router-dom";
import { useHistory } from "react-router-dom";
import "./Events.css";
import staticMsg from "./helpers/staticUserMsg";
import BoogieBotApi from "./Api";
import EventContext from "./helpers/EventContext";

const Events = ({ currentUser }) => {
	const history = useHistory();
	const { events, setEvents } = useContext(EventContext);

	let filteredEvents = currentUser.isAdmin
		? events
		: events.filter((e) => Object.values(e).includes(currentUser.username));

	const removeEvent = async (e) => {
		let btn = e.target.parentNode;
		let evtId = btn.id;

		try {
			await BoogieBotApi.removeEvent(evtId);
			let closestCard = btn.parentNode;
			closestCard.remove();
			history.push("/events");
		} catch (err) {
			console.error(err);
		}
	};

	useEffect(
		() => {
			const getEvents = async () => {
				let events = await BoogieBotApi.getEvents();
				setEvents(events);
			};

			getEvents();
		},
		[ setEvents ]
	);

	if (!events) return console.error(staticMsg.CONSOLE_EVENTS_ERROR);
	if (!currentUser.username) return console.error(staticMsg.CONSOLE_USER_ERROR);

	return (
		<div className="Events">
			<p className="Events-rmrk">
				<i>{staticMsg.EVENTS_RMRK}</i>
			</p>

			<div className="Events-body">
				<div className="Events-create">
					<Link to="/events/new" className="Events-new">
						<span className="big-bold">{staticMsg.EVENTS_CREATE}</span>
						<p>
							<i>{staticMsg.EVENTS_NEW}</i>
						</p>
					</Link>
				</div>

				<div className="Events-current">
					{filteredEvents &&
						filteredEvents.map((event) => (
							<div key={event.id}>
								<Link to={`/events/${event.id}`} key={event.id} className="link">
									<Card className="evt">
										<CardImg
											className="evt-img"
											top
											width="100%"
											src={event.imgUrl || staticMsg.DEFAULT_EVENT_IMG}
											alt="Card image cap"
										/>
										<CardBody>
											<CardTitle tag="h5">{event.title}</CardTitle>
										</CardBody>
									</Card>
								</Link>
								<div id={event.id}>
									<Button onClick={removeEvent}>Delete</Button>
								</div>
							</div>
						))}
				</div>
			</div>
		</div>
	);
};

export default Events;
