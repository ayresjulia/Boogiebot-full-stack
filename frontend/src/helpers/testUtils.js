import React from "react";
import UserContext from "./UserContext";
import EventContext from "./EventContext";

const demoUser = {
	username: "testuser",
	firstName: "testfirst",
	lastName: "testlast",
	email: "test@test.net",
	imgUrl: null
};

const testEvents = [
	{
		id: 1,
		title: "Anniversary",
		description: "Anniversary",
		eventDate: "2021-05-19",
		eventTime: "13:00",
		city: "Minsk",
		state: "CA",
		country: "Belarus",
		imgUrl:
			"https://images.unsplash.com/photo-1532117182044-031e7cd916ee?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80",
		hostUsername: "testuser"
	},
	{
		id: 2,
		title: "Wedding",
		description: "Wedding",
		eventDate: "2021-05-19",
		eventTime: "13:00",
		city: "Minsk",
		state: "CA",
		country: "Belarus",
		imgUrl:
			"https://images.unsplash.com/photo-1532117182044-031e7cd916ee?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80",
		hostUsername: "testuser"
	}
];

const UserProvider = ({ children, currentUser = demoUser }) => (
	<UserContext.Provider value={{ currentUser }}>{children}</UserContext.Provider>
);

const EventProvider = ({ children, events = testEvents }) => (
	<EventContext.Provider value={{ events }}>{children}</EventContext.Provider>
);

export { UserProvider, EventProvider };
