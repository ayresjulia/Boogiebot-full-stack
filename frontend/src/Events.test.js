import React from "react";
import { render } from "@testing-library/react";
import Events from "./Events";
import { MemoryRouter } from "react-router";
import { EventProvider } from "./helpers/testUtils";

let currentUser = { username: "testuser", isAdmin: false };

it("renders without crashing", () => {
	render(
		<MemoryRouter>
			<EventProvider>
				<Events currentUser={currentUser} />
			</EventProvider>
		</MemoryRouter>
	);
});

it("matches snapshot", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<EventProvider>
				<Events currentUser={currentUser} />
			</EventProvider>
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

it("renders create new event link", () => {
	const { getByText } = render(
		<MemoryRouter>
			<EventProvider>
				<Events currentUser={currentUser} />
			</EventProvider>
		</MemoryRouter>
	);

	expect(getByText("your events will be shown here.", { exact: false })).toBeInTheDocument();
	expect(getByText("create").parentNode.href).toBe("http://localhost/events/new");
});
