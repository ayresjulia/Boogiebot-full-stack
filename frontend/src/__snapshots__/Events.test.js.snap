// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Events"
  >
    <p
      class="Events-rmrk"
    >
      <i>
        your events will be shown here. click on event to get details or create a new one
      </i>
    </p>
    <div
      class="Events-body"
    >
      <div
        class="Events-create"
      >
        <a
          class="Events-new"
          href="/events/new"
        >
          <span
            class="big-bold"
          >
            create
          </span>
          <p>
            <i>
              new event.
            </i>
          </p>
        </a>
      </div>
      <div
        class="Events-current"
      >
        <div>
          <a
            class="link"
            href="/events/1"
          >
            <div
              class="evt card"
            >
              <img
                alt="Card image cap"
                class="evt-img card-img-top"
                src="https://images.unsplash.com/photo-1532117182044-031e7cd916ee?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80"
                width="100%"
              />
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Anniversary
                </h5>
              </div>
            </div>
          </a>
          <div
            id="1"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
        <div>
          <a
            class="link"
            href="/events/2"
          >
            <div
              class="evt card"
            >
              <img
                alt="Card image cap"
                class="evt-img card-img-top"
                src="https://images.unsplash.com/photo-1532117182044-031e7cd916ee?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=80"
                width="100%"
              />
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Wedding
                </h5>
              </div>
            </div>
          </a>
          <div
            id="2"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
